test/unit/metrics/test_annotated_metric_function.py

```diff
@@ -15,7 +16,7 @@ def test_constructor_unnamed():
     assert len(fc.kw_argument_mapping) == 0
 
 
-def test_constructor_no_name():
+def test_constructor_no_name(recwarn):
```

Comment:
I didn't know about `recwarn`, it's neat.

Reply:
Neither did I until yesterday :-)